src/model/models/modelManager.ts

```diff
@@ -61,4 +65,45 @@ export class ModelManager {
       throw new AppError('', httpStatus.INTERNAL_SERVER_ERROR, 'store-trigger service is not available', true);
     }
   }
+
+  public async deleteModel(identifier: string): Promise<StoreTriggerResponse> {
+    this.logger.debug({ msg: 'delete record', modelId: identifier });
+    try {
+      const record: Record3D | undefined = await this.catalog.getRecord(identifier);
+      if (record === undefined) {
+        this.logger.error({ msg: 'model identifier not found', modelId: identifier });
+        throw new AppError('NOT_FOUND', httpStatus.NOT_FOUND, `Identifier ${identifier} wasn't found on DB`, true);
+      }
+      if (record.productStatus != RecordStatus.UNPUBLISHED) {
+        this.logger.error({ msg: 'model with status PUBLISHED cannot be deleted', modelId: identifier });
+        throw new AppError(
+          'BAD_REQUEST',
+          httpStatus.BAD_REQUEST,
+          `Model ${record.productName} is PUBLISHED. The model must be UNPUBLISHED to be deleted!`,
+          true
+        );
+      }
+      this.logger.info({ msg: 'starting deleting record', modelId: identifier, modelName: record.productName });
+    } catch (error) {
+      if (error instanceof AppError) {
+        throw error;
+      }
+    }
+    const record: Record3D | undefined = await this.catalog.getRecord(identifier);
```

Comment:
Why do you get the record twice?

Reply:
because it is in 2 try & catchs.

Reply:
Why?

Reply:
one for the get record and one for the deletemodel. the getrecord function respone is record3d or undefined.
i cant send a request to store trigger of undefined

Reply:
I don't understand.. you are getting the same record twice. The fewer requests for external services, the better. Fix this

Reply:
one for the get record and one for the deletemodel. the getrecord function respone is record3d or undefined.
i cant send a request to store trigger of undefined